raiden/utils/cli.py

```diff
@@ -0,0 +1,230 @@
+        rows = list(rows)
+        if widths is None:
+            widths = measure_table(rows)
+        if len(widths) != 2:
+            raise TypeError('Expected two columns for definition list')
+
+        first_col = min(widths[0], col_max) + col_spacing
+
+        for first, second in iter_rows(rows, len(widths)):
+            self.write('%*s%s' % (self.current_indent, '', first))
+            if not second:
+                self.write('\n')
+                continue
+            if term_len(first) <= first_col - col_spacing:
+                self.write(' ' * (first_col - term_len(first)))
+            else:
+                self.write('\n')
+                self.write(' ' * (first_col + self.current_indent))
+
+            text_width = max(self.width - first_col - 2, 10)
+            lines = iter(wrap_text(second, text_width).splitlines())
+            if lines:
+                self.write(next(lines) + '\n')
+                for line in lines:
+                    self.write('%*s%s\n' % (
+                        first_col + self.current_indent, '', line))
+            else:
+                self.write('\n')
+
+
+class Context(click.Context):
+    def make_formatter(self):
+        return HelpFormatter(width=self.terminal_width, max_width=self.max_content_width)
+
+
+class CustomContextMixin:
+    """ Use above context class instead of the click default """
+    def make_context(self, info_name, args, parent=None, **extra):
+        """
+        This function when given an info name and arguments will kick
+        off the parsing and create a new :class:`Context`.  It does not
+        invoke the actual command callback though.
+
+        :param info_name: the info name for this invokation.  Generally this
+                          is the most descriptive name for the script or
+                          command.  For the toplevel script it's usually
+                          the name of the script, for commands below it it's
+                          the name of the script.
+        :param args: the arguments to parse as list of strings.
+        :param parent: the parent context if available.
+        :param extra: extra keyword arguments forwarded to the context
+                      constructor.
+        """
+        for key, value in iter(self.context_settings.items()):
+            if key not in extra:
+                extra[key] = value
+        ctx = Context(self, info_name=info_name, parent=parent, **extra)
+        with ctx.scope(cleanup=False):
+            self.parse_args(ctx, args)
+        return ctx
+
+
+class GroupableOption(click.Option):
+    def __init__(
+            self,
+            param_decls=None,
+            show_default=False,
+            prompt=False,
+            confirmation_prompt=False,
+            hide_input=False,
+            is_flag=None,
+            flag_value=None,
+            multiple=False,
+            count=False,
+            allow_from_autoenv=True,
+            type=None,
```

Comment:
I tried to rename this to `type_` since both `type` and `help` are reserved keywords and I got `TypeError: __init__() got multiple values for argument 'type'`. What kind of magic does this class do?

Reply:
This is an inherited interface from `click` so unfortunately you can't rename the parameters